### PR TITLE
Removed service agents dependency

### PR DESCRIPTION
**Pull request by rajendramachani**
This PR contains the following changes,
(1) Synced changes from the top of the tree patch on the local repo.
(2) Removed dependency for service vendor agents in the devstack and diskimage_create code.

### CODE REVIEW

File: devstack/lib/gbp [hunk 1]
```diff
@@ -26,6 +26,8 @@ AIM_REPO=http://github.com/noironetworks/aci-integration-module.git
 AIM_DIR=$DEST/aim
 APICML2_REPO=http://github.com/noironetworks/apic-ml2-driver.git
 APICML2_DIR=$DEST/apic_ml2
+OPFLEX_REPO=http://github.com/noironetworks/python-opflex-agent.git
+OPFLEX_DIR=$DEST/opflexagent
```
Comment (njagadish) [inline devstack/lib/gbp]: Lets not sync with GBP master now. We should do this for all code once.

File: devstack/local.conf.nfp [hunk 2]
```diff
@@ -30,8 +30,7 @@ if [[ $ENABLE_NFP = True ]]; then
         EXT_NET_CIDR=
     
         # Following image path options are optional
-        # Configure them to use the custom images
-        # Leave them empty, so that they will be built during the installation
+        # Leave ConfiguratorQcow2Image empty to let the install step build it
```
Comment (njagadish) [inline devstack/local.conf.nfp]: Specify the "optional" meaning for configurator and vyos/haproxy differently.

File: devstack/override-defaults [hunk 3]
```diff
@@ -1 +1,13 @@
 NEUTRON_CREATE_INITIAL_NETWORKS="False"
+
+ENABLE_APIC_AIM=${ENABLE_APIC_AIM:-False}
+
+if [[ $ENABLE_APIC_AIM = True ]]; then
+    echo_summary "Overriding defaults for apic_aim"
```
Comment (njagadish) [inline devstack/override-defaults]: Lets not change this.

File: devstack/plugin.sh [hunk 4]
```diff
@@ -86,6 +86,7 @@ if is_service_enabled group-policy; then
         echo_summary "Preparing $GBP"
     elif [[ "$1" == "stack" && "$2" == "install" ]]; then
         echo_summary "Installing $GBP"
+        [[ $ENABLE_APIC_AIM = True ]] && install_apic_aim
```
Comment (njagadish) [inline devstack/plugin.sh]: Lets not change this file also.

File: devstack/settings [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 source $DEST/gbp/devstack/lib/gbp
+
+[[ $ENABLE_APIC_AIM = True ]] && source $DEST/gbp/devstack/lib/apic_aim
+
 ENABLE_NFP=${ENABLE_NFP:-False}
 [[ $ENABLE_NFP = True ]] && NFP_DEVSTACK_MODE=${NFP_DEVSTACK_MODE:-base}
 [[ $ENABLE_NFP = True ]] && source $DEST/gbp/devstack/lib/nfp
 # VM locations
 ConfiguratorQcow2Image=${ConfiguratorQcow2Image:-build}
-VyosQcow2Image=${VyosQcow2Image:-build}
```
Comment (njagadish) [inline devstack/settings:8]: Retain this change and revert all other changes in this file.

**Comment by rajendramachani** [comment]
Addressed the review comments in    dbdd0ad

**Pull request merged**